charts/argo-cd/templates/dex/certificate.yaml

```diff
@@ -0,0 +1,38 @@
+{{- if and .Values.dex.enabled .Values.dex.certificate.enabled -}}
+apiVersion: {{ include "argo-cd.apiVersion.cert-manager" . }}
+kind: Certificate
+metadata:
+  name: {{ include "argo-cd.dex.fullname" . }}
+  labels:
+    {{- include "argo-cd.labels" (dict "context" . "component" .Values.dex.name "name" .Values.dex.name) | nindent 4 }}
+spec:
+  secretName: argocd-dex-server-tls
+  {{- with .Values.dex.certificate.domain }}
+  commonName: {{ . | quote }}
+  {{- end }}
+  dnsNames:
+    - {{ include "argo-cd.dex.fullname" . }}
```

Comment:
Only K8s domains argocd-dex-server and argocd-dex-server.{{ namespace }}.svc is needed